docs: document pebble enter command (#867)
canonical/pebble: docs/reference/cli-commands.md

<!-- END AUTOMATED OUTPUT FOR checks -->

(reference_pebble_enter_command)=
## enter

> The `enter` command is used to run Pebble as a container entrypoint, optionally invoking another Pebble subcommand inside the same environment.

Reviewer: The command/subcommand terminology varies throughout the doc and I don't think it's possible to make it 100% precise without becoming unreadable. But I'd go with "command" here as we're comparing to the same level as `enter`.

```suggestion
The `enter` command is used to run Pebble as a container entrypoint, optionally invoking another Pebble command inside the same environment.
```

          --identities=   Seed identities from file (like update-identities
                          --replace)
          --run           Start default services before executing subcommand
```
<!-- END AUTOMATED OUTPUT FOR enter -->

### How it works

`pebble enter` is intended to be the container's entrypoint. With no subcommand it behaves like [`pebble run`](#reference_pebble_run_command): the daemon starts and runs the configured services in the foreground until interrupted.

When a subcommand is given, `enter` runs that subcommand against the same Pebble environment, choosing the right runtime behavior for it:

* `help`, `version` — run without starting the service manager.
* `plan`, `services`, `ls` — start the service manager but skip default-service autostart (so the inspection is quick and quiet); logs are suppressed on stdout unless `-v`/`--verbose` is passed.
* `exec` — start the service manager (without autostarting default services) and run a one-shot remote command; logs are suppressed.
* `start`, `stop` — start the service manager and keep it running after the subcommand returns, so the container stays up.

> This makes `enter` a convenient way to share one container image across run, inspection and exec entrypoints without invoking the daemon separately each time.

Reviewer: I found the concept of "sharing" a container image confusing. Also, the above breakdown of subcommands suggests that the daemon does start - if I understand correctly that "service manager" == "daemon". How about something like this instead:

```suggestion
This makes `enter` a convenient way for a container image to provide inspection and exec modes that don't start the default services.
```

### Examples

Use Pebble as the container entrypoint, starting the daemon and configured services:

```{terminal}
pebble enter
```

Run a one-shot command inside the same Pebble environment used by the container:

```{terminal}
pebble enter exec -- echo hello
```

Inspect the plan without starting any services:

```{terminal}
pebble enter plan

> ```

Author: ```suggestion
```

> ```

Reviewer: Would it be better to (also?) show examples from outside the container? Such as:

```
docker run --rm <image> plan
```

Author: Discussed this in a call and we feel that we should just drop the whole examples section.
